README.md

```diff
@@ -39,48 +39,84 @@ In particular, please read the [Pull Request Guidelines](https://github.com/exer
-implementation. Navigate into the *leap* directory, you'll see there are a
-number of files there.
+Let's walk through the an imaginary `exercise-slug` exercise to see what is included in an exercise
+implementation. In any exercise you may see a number of files present:
 
 ```sh
-~/exercism/go/leap
+~/exercism/go/exercise-slug
 $ tree -a
 .
 ├── cases_test.go
 ├── example.go
-├── leap.go
-├── leap_test.go
-└── .meta
-    └── gen.go
-
-1 directory, 5 files
+├── exercise_slug.go
+├── exercise_slug_test.go
+├── .meta
+│   └── description.md
+│   └── gen.go
+│   └── hints.md
+│   └── metadata.yml
+└── README.md
 ```
 
-This list of files can vary across exercises so let's quickly run through
-each file and briefly describe what it is.
-
-* **cases_test.go** - This file contains [generated test cases](#generating-test-cases),
-    and will only be present in some exercises.
-
-* **example.go** - This is a reference solution for the exercise. This file is
-    ignored by the `exercism fetch` command. See [ignored files](#ignored-files) for
-    details on which files are ignored.
+This list of files *can vary* across exercises. Not all exercise
+utilize all of the available files, but they are available for use
+should an exercise need to adjust itself against [the global problem-specification repository](https://github.com/exercism/problem-specifications/tree/master/exercises). This
+repository helps collect common information for all exercises across
+all tracks. However, should local track-specific modifications or
+enhancements be made with respect to the global problem specifications
+they may be performed by the presence of specifically named files in
+an exercises `.meta/` directory.
 
-* **leap.go** - This is a *stub file*, and will only be present in some exercises.
+So let's quickly run through each potential file and briefly describe
+what and why it is.
 
-* **leap_test.go** - This is the main test file for the exercise.
-
-* **.meta/** - The *.meta* directory contains files that are not meant to be
-    included when a user fetches an exercise, for instance test case generators.
-    See [ignored files](#ignored-files) for details on which files are ignored.
-
-* **gen.go** - This file, within the *.meta* directory, generates the *cases_test.go*
-    file, and will only be present in some exercises. See [generating test cases](#generating-test-cases)
-    for more information.
-
-In some exercises there can be extra files, for instance the `series` exercise
-contains extra test files.
+* **cases_test.go** - This file contains [generated test cases](#generating-test-cases),
+  and will only be present in some exercises. These will be 
+  automatically generated by a *.meta/gen.go* file and should
+  represent test data sourced from the problem-specifications repository.
+
+* **example.go** - This is an [example solution](#example-solutions) for
+  the exercise. It is used in integration testing and to provide a
```

Comment:
*It is used in integration testing* - is it?

Maybe: *It is used ensure the test suite is correct, and to provide a reference solution to the exercise.*

Reply:
Yep it is used to make sure the test suite passes the Travis build.